server/fleet/software.go

```diff
@@ -162,6 +162,30 @@ type SoftwareTitle struct {
 	SoftwarePackage *SoftwareInstaller `json:"software_package" db:"-"`
 }
 
+// This type is essentially the same as the above SoftwareTitle type. The only difference is that
```

Comment:
I _think_ this could work by embedding the `SoftwareTitle` struct inside `SoftwareTitleListResult` and just having an explicit `SoftwarePackage` field that "overrides" the one in `SoftwareTitle`. But I'm not sure if all third-party packages we use that depend on struct tags would support that properly (`json` would, I'm pretty sure). Anyway, more an FYI than a blocker to merge, up to you if you want to try it out or not.